adapters/backend/v1/adapter.go

```diff
@@ -191,59 +192,98 @@ func (b *Adapter) Batch(ctx context.Context, kind domain.Kind, batchType domain.
 	return client.Batch(ctx, kind, batchType, items)
 }
 
+type ReconciliationTask struct {
+	ctx     context.Context
+	cfg     *config.ReconciliationTaskConfig
+	adapter *Adapter
+}
+
 // startReconciliationPeriodicTask starts a periodic task that sends reconciliation request messages to connected clients
 // every configurable minutes (interval). If interval is 0 (not set), the task is disabled.
+// when cron schedule is set, the task will be executed according to the cron schedule.
 // intervalFromConnection is the minimum interval time in minutes from the connection time that the reconciliation task will be sent.
 func (a *Adapter) startReconciliationPeriodicTask(mainCtx context.Context, cfg *config.ReconciliationTaskConfig) {
-	if cfg == nil || cfg.TaskIntervalSeconds == 0 || cfg.IntervalFromConnectionSeconds == 0 {
+	if cfg == nil || (cfg.TaskIntervalSeconds == 0 && cfg.CronSchedule == "") || cfg.IntervalFromConnectionSeconds == 0 {
 		logger.L().Warning("reconciliation task is disabled (intervals are not set)")
 		return
 	}
 
-	go func() {
-		logger.L().Info("starting reconciliation periodic task",
-			helpers.Int("TaskIntervalSeconds", cfg.TaskIntervalSeconds),
-			helpers.Int("IntervalFromConnectionSeconds", cfg.IntervalFromConnectionSeconds))
-		ticker := time.NewTicker(time.Duration(cfg.TaskIntervalSeconds) * time.Second)
-		for {
-			select {
-			case <-mainCtx.Done():
-				ticker.Stop()
+	task := ReconciliationTask{
+		ctx:     mainCtx,
+		cfg:     cfg,
+		adapter: a,
+	}
+
+	if cfg.CronSchedule != "" {
```

Comment:
I would instantiate another ticker instead of rewriting a new branch of code.
Please take a look at https://github.com/krayzpipes/cronticker/blob/main/cronticker/ticker.go (maybe we can just copy/paste this ticker and use the other cron package for our needs)

Reply:
LGTM

Reply:
Can we release?